pkg/messageformat/lex.go

```diff
@@ -211,12 +211,13 @@ func lexExpr(l *lexer) stateFn {
 		}
 	case '$':
 		return lexVariable(l)
+
 	case ':':
 		return lexFunction(l)
 	case '+':
-		return lexOpeningFunction(l)
+		return processFunction(l, v)
```

Comment:
No need for `processFunction`, it makes it harder to read the code, and slower, too.

One option is ...

```suggestion
		if l.exprDepth > 1 {
			return lexOpeningFunction(l)
		}
		
		l.backup()
		
		return lexText(l)
```